### PR TITLE
Support GHC 9.2.

### PR DESCRIPTION
**Pull request by patrickt**
Widens template-haskell bounds and removes `mappend` definitions in favor of defaulting to `<>`, as per `-Wnoncanonical-monoid-instances`.

### CODE REVIEW

**Comment by patrickt** [comment]
This is ready for review.

**Comment by patrickt** [comment]
@jacobstanley ping

**Comment by weebl2000** [comment]
Cool.

**Comment by tmcgilchrist** [comment]
Could you add this GHC version to the GH CI setup so it gets fully tested.
https://github.com/hedgehogqa/haskell-hedgehog/blob/master/.github/workflows/ci.yaml#L17

Seems like a maintainer needs to approve the PR so CI gets kicked off.

**Comment by patrickt** [comment]
@jacobstanley Any word on this?

**Comment by smunix** [comment]
+ @moodmosaic

**Review by moodmosaic — APPROVED** [review]
👍 🚀 🍻

**Comment by moodmosaic** [comment]
It looks like some builds are failing. Have to investigate on that.

**Comment by gwils** [comment]
It looks as though you will need to use CPP to give `mappend` an explicit definition on GHC 8.2 and 8.0, in order to make `-Werror` happy.

**Comment by moodmosaic** [comment]
> It looks as though you will need to use CPP to give `mappend` an explicit definition on GHC 8.2 and 8.0, in order to make `-Werror` happy.

Could you do that, @patrickt? Otherwise we'll get to it as soon as possible.

**Comment by patrickt** [comment]
@moodmosaic On it!

**Comment by ysangkok** [comment]
@patrickt It is better to define `mappend = (<>)` on versions of GHC where Semigroup is a superclass of Monoid. Otherwise this warning is triggered: https://downloads.haskell.org/~ghc/9.2.1/docs/html/users_guide/using-warnings.html#ghc-flag--Wnoncanonical-monoid-instances

**Comment by patrickt** [comment]
@moodmosaic I’ve tested this with 8.2 and 8.0 locally; can I get another approval/merge if it looks good to you?

**Comment by moodmosaic** [comment]
> @moodmosaic I’ve tested this with 8.2 and 8.0 locally; can I get another approval/merge if it looks good to you?

Sure, I just run it. 🍻

**Comment by moodmosaic** [comment]
@patrickt, some are still failing but we're almost there.

**Comment by moodmosaic** [comment]
@patrickt, some are still failing but we're almost there.

**Comment by patrickt** [comment]
@moodmosaic Ugh, my bad. This last one should be okay.

**Comment by moodmosaic** [comment]
> @moodmosaic Ugh, my bad. This last one should be okay.

No problem! Hmm, I think there's one more left to go 👀

**Comment by moodmosaic** [comment]
@patrickt, I know this is getting boring at this point 😴 It looks like some checks are still failing.

**Comment by moodmosaic** [comment]
Yep, looking at the build error it looks like changes in https://github.com/hedgehogqa/haskell-hedgehog/pull/436/commits/94dd5b81e46d4a636521acf08039c8e3d848ef37 must also be applied in `Registry.hs`. @patrickt, let me know if you'd want me to do that.

**Comment by ysangkok** [comment]
@patrickt looks like a new warning in 9.2 makes the build fail:

```
src\Test\Example\Resource.hs:174:7: error: [-Wincomplete-uni-patterns, -Werror=incomplete-uni-patterns]
Error:     Pattern match(es) are non-exhaustive
    In a pattern binding:
        Patterns of type ‘([String], [String])’ not matched:
            ([], [])
            ((_:_), [])
    |
174 |       (xs, x : ys) =

    |       ^^^^^^^^^^^^^^^...
```

**Comment by patrickt** [comment]
Sorry for the churn, everyone. I’ve turned on `-Werror` locally and observed a clean build. Hopefully we’re good to go.

**Comment by gwils** [comment]
Perhaps a maintainer could trigger a CI build to verify this? @moodmosaic

**Comment by moodmosaic** [comment]
> Sorry for the churn, everyone. I’ve turned on `-Werror` locally and observed a clean build. Hopefully we’re good to go.

No problem at all! 🍻

**Comment by moodmosaic** [comment]
> Perhaps a maintainer could trigger a CI build to verify this? @moodmosaic

@gwils, CI build started 🚀

**Comment by moodmosaic** [comment]
Thank you, @patrickt 🍻 Also thank you @ysangkok, @gwils, @tmcgilchrist for the great feedback 🍺

**Pull request merged**
